### PR TITLE
chore: add deploy job to test and deploy gh workflow

### PR DESCRIPTION
**Pull request by Hunga1**
Added a deploy job to the test and deploy GitHub Action workflow that creates a new GitHub release when run for tags. Updated build status badge and Slack notification to cover deployments by the workflow.

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guidelines](https://github.com/sendgrid/sendgrid-php/blob/main/CONTRIBUTING.md) and my PR follows them
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the main branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary documentation about the functionality in the appropriate .md file
- [x] I have added inline documentation to the code I modified

### CODE REVIEW

File: .github/workflows/test-and-deploy.yml [hunk 1]
```diff
@@ -29,18 +30,28 @@ jobs:
       - name: Run Unit Tests
         run: make test-docker version=${{ matrix.php }}
 
+    deploy:
+      name: Deploy
+      if: success() && github.ref_type == 'tag'
+      needs: [ test ]
+      steps:
+        - name: Create GitHub Release
+        uses: sendgrid/dx-automator/actions/release@main
```
Comment (childish-sambino) [inline .github/workflows/test-and-deploy.yml:39]: nit: alignment is off.

File: .github/workflows/test-and-deploy.yml [hunk 2]
```diff
@@ -29,18 +30,28 @@ jobs:
       - name: Run Unit Tests
         run: make test-docker version=${{ matrix.php }}
 
+    deploy:
+      name: Deploy
+      if: success() && github.ref_type == 'tag'
+      needs: [ test ]
+      steps:
+        - name: Create GitHub Release
```
Comment (childish-sambino) [inline .github/workflows/test-and-deploy.yml:38]: Also need to include the `sendgrid-php.zip` asset. It's generated with `make bundle`: https://github.com/sendgrid/sendgrid-php/blob/1daa621985dd49da0471485f96dab4ab0e6338c6/Makefile#L41

File: .github/workflows/test-and-deploy.yml (hunk 1 above)
Comment (Hunga1) [reply .github/workflows/test-and-deploy.yml:45]: Fixed

File: .github/workflows/test-and-deploy.yml (hunk 2 above)
Comment (Hunga1) [reply .github/workflows/test-and-deploy.yml]: Added the steps to install and and build the library artifact

File: .github/workflows/test-and-deploy.yml [hunk 3]
```diff
@@ -29,18 +30,37 @@ jobs:
...
+    name: Deploy
+    if: success() && github.ref_type == 'tag'
+    needs: [ test ]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sendgrid-php
+        uses: actions/checkout@v2
+
+      - name: Build Release Artifacts
+        run: make bundle
```
[the remark this answers is not in the document]
Comment (Hunga1) [reply .github/workflows/test-and-deploy.yml:43]: Yes, that's right. Good catch. Should the build environment use the latest supported PHP version (7.4)?

[the remark this answers is not in the document]
Comment (childish-sambino) [reply .github/workflows/test-and-deploy.yml:43]: Yup.

[the remark this answers is not in the document]
Comment (Hunga1) [reply .github/workflows/test-and-deploy.yml:49]: Done

**Review by childish-sambino — APPROVED** [review]

**Pull request merged**